code/modules/station_goals/bluespace_tap.dm

```diff
@@ -316,8 +317,9 @@
 
 
 /obj/machinery/power/bluespace_tap/connect_to_network()
-	..()
+	var/to_return = ..()
 	update_icon()
+	return to_return
```

Comment:
```suggestion
	. = ..()
	if(.)
		update_icon()
```
BYOND has a native return variable called `.` which will automatically return at the end of the proc.

Reply:
i added that change and verified the icon updates as expected